notes/lecture22.tex

```diff
@@ -0,0 +1,205 @@
+
+
+\section{Lecture 22: Non-convex constraints} 
+
+\begin{center}
+Seyed Mehran Mirramezani and Serena Yuan
+\end{center}
+
+
+Last time we discussed optimization of a convex function $f$ over non-convex sets. A typical example for such a problem is $\min \| Ax-y\|_2^2$ subject to $\|x\|_0 \leq s$. Another type of non-convex optimization is to minimize a non-convex function over a convex set. A typical example for this kind of non-convex optimization is known as $l_0$-minimization where $\min \| x\|_0$ subject to an affine convex constraint like $Ax=y$. In sum, the $l_0$-minimization is: 
+
+$$\min_{Ax=y} \|x\|_0$$
+
+One option for solving the above non-convex optimization problem is to relax the $l_0$-objective to the convex $l_1$-objective as:
+
+$$\min_{Ax=y} \|x\|_1$$
+
+A variation of the above problem is Basis Pursuit Denoising (BPDN),
```

Comment:
I thought about this a little and decided that the variants of l1 minimization are probably better covered in Lecture 21 because the other material on convex relaxation is there as well (you're right though that in class I only mentioned it in Lecture 22).

Since BPDN etc are also mentioned in Lecture 21 now (see below) , I think we can remove them here

https://github.com/ee227c/ee227c.github.io/blob/master/notes/lecture21.tex